File: DRRR.Server/Dtos/ChatRoomDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DRRR.Server.Dtos
{
    /// <summary>
    /// 聊天室信息DTO
    /// </summary>
    public class ChatRoomDto
    {
        /// <summary>
        /// ID
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// 房间名
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// 房主名
        /// </summary>
        public string OwnerName { get; set; }

        /// <summary>
        /// 最大用户数
        /// </summary>
        public int MaxUsers { get; set; }

        /// <summary>
        /// 当前用户数
        /// </summary>
        public int CurrentUsers { get; set; }

        /// <summary>
        /// 是否为加密房
        /// </summary>
        public bool IsEncrypted { get; set; }

        /// <summary>
        /// 密码
        /// </summary>
        public string Password { get; set; }

        /// <summary>
        /// 是否为永久房
        /// </summary>
        public bool IsPermanent { get; set; }

        /// <summary>
        /// 是否为隐藏房
        /// </summary>
        public bool IsHidden { get; set; }

        /// <summary>
        /// 是否允许游客进入
        /// </summary>
        public bool AllowGuest { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        public long CreateTime { get; set; }
    }
}

File: DRRR.Client/src/app/chat-rooms/dtos/chat-room.dto.ts
/**
 * 聊天室信息DTO
 */
export interface ChatRoomDto {
  /**
   * ID
   */
  id?: string;

  /**
   * 房间名
   */
  name?: string;

  /**
   * 房主名
   */
  ownerName?: string;

  /**
   * 最大用户数
   */
  maxUsers?: string;

  /**
   * 当前用户数
   */
  currentUsers?: string;

  /**
   * 是否为加密房
   */
  isEncrypted?: boolean;

  /**
   * 密码
   */
  password?: string;

  /**
   * 是否为永久房
   */
  isPermanent?: boolean;

  /**
   * 是否为隐藏房
   */
  isHidden?: boolean;

  /**
   * 是否允许游客进入
   */
  allowGuest?: boolean;

  /**
   * 创建时间
   */
  createTime?: number;
}
